src/main/java/org/opensrp/service/PractitionerRoleService.java

```diff
@@ -114,4 +114,17 @@ public void deletePractitionerRole(String organizationIdentifier, String practit
         return getPractitionerRoleRepository().getPgRolesForPractitioner(practitionerIdentifier);
     }
 
+    public void assignPractitionerRole(Long organizationId, String practitionerIdentifier, String code,
+            PractitionerRole practitionerRole) {
+        validateIdentifier(practitionerRole.getIdentifier());
+        Long practitionerId = practitionerService.getPractitionerIdByIdentifier(practitionerIdentifier);
+        practitionerRoleRepository
+                .assignPractitionerRole(organizationId, practitionerId, practitionerIdentifier, code, practitionerRole);
+    }
+
+    public void validateIdentifier(String identifier) {
+        if (StringUtils.isBlank(identifier))
+            throw new IllegalArgumentException("Practitioner Role Identifier not specified");
```

Comment:
I think the message is wrong

Reply:
It is called inside a method assignPractitionerRole(Long organizationId, String practitionerIdentifier, String code,            PractitionerRole practitionerRole)  therefore it verifies whether a practitionerRole has an identifier or not.